File: CS/DXApplication206/CustomGridViewInfoRegistrator.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace DXApplication206 {
    public class CustomGridViewInfoRegistrator : GridInfoRegistrator {
        public override string ViewName {
            get {
                return "CustomGridView";
            }
        }

        public override BaseView CreateView(GridControl grid) {
            return new CustomGridView(grid);
        }

        public override BaseViewInfo CreateViewInfo(BaseView view) {
            return new CustomGridViewInfo(view as CustomGridView);
        }
    }
}

File: VB/DXApplication206/CustomGridViewInfoRegistrator.vb
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace DXApplication206
    Public Class CustomGridViewInfoRegistrator
        Inherits GridInfoRegistrator

        Public Overrides ReadOnly Property ViewName() As String
            Get
                Return "CustomGridView"
            End Get
        End Property

        Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
            Return New CustomGridView(grid)
        End Function

        Public Overrides Function CreateViewInfo(ByVal view As BaseView) As BaseViewInfo
            Return New CustomGridViewInfo(TryCast(view, CustomGridView))
        End Function
    End Class
End Namespace